### PR TITLE
Unify the ext_localconf.php structure

### PR DESCRIPTION
**Pull request by linawolf**
references https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/2782

### CODE REVIEW

File: Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+use MyVendor\MyExtension\Database\Query\Restriction\CustomRestriction;
+
+defined('TYPO3') or die();
+
+if (!isset($GLOBALS['TYPO3_CONF_VARS']['DB']['additionalQueryRestrictions'][CustomRestriction::class])) {
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php:9]: The whole can be shortened with:

```php
$GLOBALS['TYPO3_CONF_VARS']['DB']['additionalQueryRestrictions'][CustomRestriction::class] ??= [];
```

File: Documentation/ApiOverview/Fluid/Syntax.rst [hunk 2]
```diff
@@ -153,12 +153,9 @@ There are 3 ways to import ViewHelper namespaces in TYPO3. In all three examples
...
 
-   .. code-block:: php
-      :caption: EXT:blog_example/ext_localconf.php
-
-       $GLOBALS['TYPO3_CONF_VARS']['SYS']['fluid']['namespaces']['blog'] = [
-           'MyVendor\BlogExample\ViewHelpers',
-       ];
+    ..  literalinclude:: _Syntax/_ext_localconf.php
+        :language: php
+        :caption: EXT:mye_extension/ext_localconf.php
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Fluid/Syntax.rst:158]:
```suggestion
        :caption: EXT:my_extension/ext_localconf.php
```

File: Documentation/ApiOverview/Routing/_AdvancedRoutingConfiguration/_ext_localconf.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+use MyVendor\MyPackage\Routing\CustomEnhancer;
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Routing/_AdvancedRoutingConfiguration/_ext_localconf.php:5]:
```suggestion
use MyVendor\MyExtension\Routing\CustomEnhancer;
```

As the caption refers to "my_extension".

File: Documentation/ApiOverview/Rte/InTheBackend/PlugRte.rst [hunk 4]
```diff
@@ -26,95 +24,25 @@ API for rich text editors
...
+    ..  literalinclude:: _RichTextNodeResolver.php
+        :language: php
+        :caption: :php:`\MyVendor\MyExtension\Form\Resolver\RichTextNodeResolver`
 
-- Next step is to implement the RichtTextElement class. You can look up the
-  code of :t3src:`rte_ckeditor/Classes/Form/Element/RichTextElement.php`, which
-  does the same for ckeditor. What basically happens in its render() function,
-  is to apply any settings from the fields TCA config and then printing out all
-  of the html markup and javascript necessary for booting up the ckeditor.
+-  Next step is to implement the RichtTextElement class. You can look up the
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Rte/InTheBackend/PlugRte.rst:44]:
```suggestion
-  Next step is to implement the RichTextElement class. You can look up the
```

File: Documentation/ApiOverview/Rte/InTheBackend/_RichTextNodeResolver.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function __construct(NodeFactory $nodeFactory, array $data)
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * Returns RichTextElement as class name if RTE widget should be rendered.
+     *
+     * @return string|void New class name or void if this resolver does not change current class name.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Rte/InTheBackend/_RichTextNodeResolver.php:33]:
```suggestion
     * @return string|null New class name or null if this resolver does not change current class name.
```

File: Documentation/ApiOverview/Rte/InTheBackend/_RichTextNodeResolver.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        ) {
+            return RichTextElement::class;
+        }
+        return null;
+    }
+
+    /**
+     * @return BackendUserAuthentication
+     */
+    protected function getBackendUserAuthentication()
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Rte/InTheBackend/_RichTextNodeResolver.php:60]:
```suggestion
    private function getBackendUserAuthentication(): BackendUserAuthentication
```

File: Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+defined('TYPO3') or die();
+
+// Add your code here
+ExtensionManagementUtility::doSomething();
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php:10]: ExtensionManagementUtility does not have a doSomething method?

File: Documentation/ExtensionArchitecture/FileStructure/ExtLocalconf.rst [hunk 8]
```diff
@@ -90,48 +90,29 @@ Extension. It does not need to be registered anywhere but will be loaded
...
-
-.. code-block:: php
-   :caption: EXT:site_package/Configuration/TCA/Overrides/pages.php
-
-   ExtensionManagementUtility::registerPageTSConfigFile();
+Put all page TSconfig that must always be loaded into file
+:file:`EXT:some_extension/Configuration/page.tsconfig`. If your extension should
+also be compatible with TYPO3 v11, you can additionally load it in the
+:file:`ext_localconf.php`: :ref:`lobal page TSconfig, compatible
+with TYPO3 11 and 12 <t3tsconfig:page-tsconfig-v11-v12>`:
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/FileStructure/ExtLocalconf.rst:111]:
```suggestion
with TYPO3 v11 and v12 <t3tsconfig:page-tsconfig-v11-v12>`:
```

File: Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+declare(strict_types=1);
+
+// all use statements must come before the definition check
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+// Prevent Script from being called directly
+defined('TYPO3') or die();
+
+// Add your code here
+ExtensionManagementUtility::doSomething();
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php:12]: Also here: ExtensionManagementUtility does not have a doSomething method?

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php (hunk 7 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php:10]: Nope. But anyone who copies this line of code should notice that he needs to exchange this with whatever he wants to do?

If I put nothing then the use statement gets removed by the cgl and we want to demonstrate in this very code snippet where to put the use statements and where to put the code that does something.

File: Documentation/ExtensionArchitecture/FileStructure/ExtLocalconf.rst [hunk 10]
```diff
@@ -90,48 +90,29 @@ Extension. It does not need to be registered anywhere but will be loaded
...
-Default page TSconfig can be added inside :file:`ext_localconf.php`, see
-:ref:`t3tsconfig:pagesettingdefaultpagetsconfig`:
-
-.. code-block:: php
-   :caption: EXT:site_package/ext_localconf.php
-
-   //use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+..  versionchanged:: 12.0
+    Page TSconfig in a file :file:`EXT:some_extension/Configuration/page.tsconfig`
+    is loaded globally.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ExtensionArchitecture/FileStructure/ExtLocalconf.rst:105]: the page ts config is loaded no matter what page you are on and even on page 0, therfore globally

File: Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php (hunk 9 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php:12]: same answer as above

File: Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php:9]: ok, will do

File: Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php (hunk 9 above)
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php:12]: same suggestion as above ;-)

**Comment by linawolf** [comment]
@brotkrueml please review again

File: Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php:9]: Hmm, don't see the change?

File: Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php (hunk 7 above)
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/BestPractises/_ext_localconf.php:10]: Sorry to bother, but see no change or reaction?

File: Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php (hunk 9 above)
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/FileStructure/_ext_localconf.php:12]: Sorry to bother, see no change or reaction?

File: Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/Database/RestrictionBuilder/_ext_localconf.php]: Sorry, pushing failed. Now the change should be visible

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**